src/js/stores/MesosStore.js

```diff
@@ -0,0 +1,221 @@
+
+const FILES_TTL = 60000;
+const STATE_TTL = 60000;
+const MASTER_ID = "master";
+
+var stateMap = {};
+var taskFileMap = {};
+var taskFileRequestQue = [];
+
+var MesosStore = Object.assign({
+
+  getState: function (nodeId) {
+    var data = stateMap[nodeId];
+    if (data == null || data.timestamp + STATE_TTL < Date.now()) {
+      invalidateState(nodeId);
+      return null;
+    }
+    return data.state;
+  },
+
+  getTaskFiles: function (taskId) {
+    var data = taskFileMap[taskId];
+    if (data == null || data.timestamp + FILES_TTL < Date.now()) {
+      invalidateTaskFiles(taskId);
+      return null;
+    }
+    return data.files;
+  }
+
+}, EventEmitter.prototype);
+
+function addState(nodeId, state) {
+  stateMap[nodeId] = {
+    state: state,
+    timestamp: Date.now()
+  };
+  MesosStore.emit(MesosEvents.CHANGE);
+}
+
+function invalidateState(nodeId) {
+  stateMap[nodeId] = null;
+}
+
+function addTaskFiles(taskId, files) {
+  taskFileMap[taskId] = {
+    files: files,
+    timestamp: Date.now()
+  };
+  MesosStore.emit(MesosEvents.CHANGE);
+}
+
+function invalidateTaskFiles(taskId) {
+  delete taskFileMap[taskId];
+}
+
+function getNodeUrl(nodeId) {
+  var master = MesosStore.getState(MASTER_ID);
+  if (master == null) {
+    throw new Error(
+      `Unable to get the node url as the master state is undefined.`
+    );
+  }
+
+  let agent = master.slaves.find((slave) => {
+    return slave.id === nodeId;
+  });
+
+  if (agent == null) {
+    throw new Error(
+      `Can't get the node url as the he node (${nodeId}) is undefined.`
+    );
+  }
+
+  let pid = agent.pid;
+  let hostname = agent.hostname;
+  return `//${hostname}:${pid.substring(pid.lastIndexOf(":") + 1)}`;
+}
+
+function getExecutorDirectory(agentId, frameworkId, taskId) {
+  var agentState = MesosStore.getState(agentId);
+
+  if (agentState == null) {
+    throw new Error(
+      `Unable get the executor directory as the agent (${agentId})
+      state is undefined.`
+    );
+  }
+
+  function matchFramework(framework) {
+    return frameworkId === framework.id;
+  }
+
+  let framework =
+    agentState.frameworks.find(matchFramework) ||
+    agentState.completed_frameworks.find(matchFramework);
+
+  if (!framework) {
+    throw new Error(
+      `Can't get the executor directory as the Framework (${frameworkId})
+       does not exist on the given agent (${agentId}).`
+    );
+  }
+
+  function matchExecutor(executor) {
+    return taskId === executor.id;
+  }
+
+  var executor =
+    framework.executors.find(matchExecutor) ||
+    framework.completed_executors.find(matchExecutor);
+
+  if (executor == null) {
+    throw new Error(
+      `Unable to get the executor directory as the task (${taskId})
+      does not exists.`
+    );
+  }
+
+  return executor.directory;
+}
+
+function resolveTaskFileRequests() {
+  var info = InfoStore.info;
+
+  // Get the marathon config, to retrieve the mesos leader (master)  url
```

Comment:
Could you remove these comments. They should not be needed.
The code should be self-explanatory, otherwise the code should be improved/refactored.